core/src/main/java/com/netflix/conductor/core/execution/mapper/SimpleTaskMapper.java

```diff
@@ -68,18 +67,15 @@ public List<TaskModel> getMappedTasks(TaskMapperContext taskMapperContext)
         WorkflowModel workflowModel = taskMapperContext.getWorkflowModel();
         int retryCount = taskMapperContext.getRetryCount();
         String retriedTaskId = taskMapperContext.getRetryTaskId();
-
-        TaskDef taskDefinition =
-                Optional.ofNullable(workflowTask.getTaskDefinition())
-                        .orElseThrow(
-                                () -> {
-                                    String reason =
-                                            String.format(
-                                                    "Invalid task. Task %s does not have a definition",
-                                                    workflowTask.getName());
-                                    return new TerminateWorkflowException(reason);
-                                });
-
+        TaskDef taskDefinition = workflowTask.getTaskDefinition();
```

Comment:
This case seems to be addressed [in MetadataMapperService](https://github.com/Netflix/conductor/blob/main/core/src/main/java/com/netflix/conductor/core/metadata/MetadataMapperService.java#L120-L133). Is there a scenario where this exception happens?